src/data.jl

```diff
@@ -47,6 +47,10 @@ function Base.length(m::MB)
     m.partial ? ceil(Int,n) : floor(Int,n)
 end
 
+function Base.rand(m::MB)
+    i = rand(0:(m.length-m.batchsize))
+    return next(m, i)[1]
+end
```

Comment:
Why do we need a rand(m::MB) ?  minibatches already have a shuffle option.

Reply:
I think it's been used in [here](https://github.com/CarloLucibello/Knet.jl/blob/6182c4f1c6b77ab84613c916cfa6023ff631653f/examples/variational-autoencoder/vae_mnist.jl#L97) to sample a random batch. Nonetheless, the following is equivalent,

```julia
x, _ = first(minibatch(xtrn, ytrn, batchsize, shuffle=true))
```